### PR TITLE
workflows: add snap_store_integration_tests to handle snap store downloads and testing

### LINKED ISSUE #142: The store snap is not tested on device - autoconnection not tested etc
We could add a way to run the snap integration tests using the store snap installed from a specific channel/version (edge/beta/candidate/release) e.g. for use before manual promotion.

This would use spread-plus.

### PR DESCRIPTION
**Pull request by Sinan-Karakaya**
This PR closes #142.

- Added `snap_store_integration_tests.yml` to handle testing the snap store version of fpgad on a specific channel, on a specific CID
- Modified `test_snap_device_script.sh` to handle installing snaps from a local snap file, or by reading snap_channel.txt and download fpgad from the snapstore

### CODE REVIEW

**Review by artiepoole — COMMENTED** [review]
LGTM in principle, just want to discuss options first

**Review by artiepoole — CHANGES_REQUESTED** [review]
I like this approach a lot more, thanks! I just have one change to make it easier to follow, and one question I answered myself - keeping for clarity later.

**Review by artiepoole — CHANGES_REQUESTED** [review]
Sorry spotted this based on the failed run

**Review by artiepoole — APPROVED** [review]
LGTM! Thanks very much Sinan. Please wait for @talhaHavadar to review before merging :)

**Review by artiepoole — CHANGES_REQUESTED** [review]

**Review by talhaHavadar — CHANGES_REQUESTED** [review]

**Review by artiepoole — DISMISSED** [review]
LGTM, thanks again Sinan!

**Review by talhaHavadar — CHANGES_REQUESTED** [review]

File: .github/testflinger-assets/test_snap_device_script.sh [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
+if [[ -f ./env_setup.sh ]]; then
+    # shellcheck disable=SC1091
```
[the remark this answers is not in the document]
Comment (Sinan-Karakaya) [reply .github/testflinger-assets/test_snap_device_script.sh]: my bad I thought shellcheck was in the linter. That rules is just to specify to the linter it should not try to lint the file im referencing. I'll just remove it since its not in the linter, and ill also remove the if statement based on your other comments

File: .github/testflinger-assets/testflinger_job.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ test_data:
 
     # Run the test script, but always retrieve artifacts even if it fails
     set +e
-    _run '/home/ubuntu/device_script.sh'
+    _run '/home/ubuntu/env_setup.sh; /home/ubuntu/device_script.sh'
```
Comment (talhaHavadar) [inline .github/testflinger-assets/testflinger_job.yaml]: ok I see in env setup we are not exporting environment variables so if we remove `. ./env_setup.sh` line in device_script.sh this would have no affect at all.

Did you try exporting environment variables in env_setup.sh and remove sourcing of env_setup in test_snap_device_script.sh (line 4) ? not sure how the scripts are being encapsulated by `_run` but if env vars are properly exported then we dont need to source env_setup.sh in device_script.sh file

Then you can have oneline for _run env_setup.sh and other line for _run device_script.sh

**Review by talhaHavadar — CHANGES_REQUESTED** [review]

File: .github/testflinger-assets/testflinger_job.yaml (hunk 2 above)
Comment (artiepoole) [reply .github/testflinger-assets/testflinger_job.yaml]: I don't know if this answers the question about exporting variables, but they are starting a new ssh shell in _run each time.
```
$SSHPASS ssh $SSH_OPTS $USER@$DEVICE_IP "export PATH=\"\$PATH:/home/$USER/$TOOLS_PATH_DEVICE\"; $@"
```

Comment (Sinan-Karakaya) [reply .github/testflinger-assets/testflinger_job.yaml]: I believe SSH session isolation won't make the environment variables available in the second `_run` command. I think just removing the call to `env_setup.sh` in `device_script.sh` and keeping the sourcing + calling as a one-liner would be the best solution. (The new commit fixes that, let me know if something is still missing)

**Review by talhaHavadar — DISMISSED** [review]
Looks good thank you Sinan, we can merge this once CI is green

**Comment by artiepoole** [comment]
Sorry for the hassle Sinan, but as you can see in #198 we changed the testing process quite a lot, so the renamed file has the wrong contents now and the new file you made doesn't do the right thing any more!
